src/main/java/com/google/devtools/build/lib/bazel/repository/downloader/HttpDownloader.java

```diff
@@ -136,4 +128,39 @@ public void download(
       throw exception;
     }
   }
+
+  /** Downloads the contents of one URL and reads it into a byte array. */
+  public byte[] downloadAndReadOneUrl(
+      URL url, ExtendedEventHandler eventHandler, Map<String, String> clientEnv)
+      throws IOException, InterruptedException {
+    HttpConnectorMultiplexer multiplexer = setUpConnectorMultiplexer(eventHandler, clientEnv);
+
+    ByteArrayOutputStream out = new ByteArrayOutputStream();
+    SEMAPHORE.acquire();
+    try (HttpStream payload = multiplexer.connect(url, Optional.absent())) {
+      ByteStreams.copy(payload, out);
+    } catch (SocketTimeoutException e) {
+      // SocketTimeoutExceptions are InterruptedIOExceptions; however they do not signify
+      // an external interruption, but simply a failed download due to some server timing
+      // out. So rethrow them as ordinary IOExceptions.
+      throw new IOException(e);
+    } catch (InterruptedIOException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)